### ISSUE TITLE
Repeated measures analysis

### ISSUE THREAD

**Issue by CrumpLab** [CrumpLab/EntropyTyping#16]
For the typing data we should do some kind of repeated measures analysis to show that there are effects of position and word length.

Problem is our design is unbalanced (position and word length are not factorials crossed)

One solution is the treat the design as a one-way analysis with 45 levels (one for each position across word lengths)

I did this, and I think Walter tried something simulate, but it breaks aov, too many subjects.

Task: find some alternative ways to do the repeated measures analysis faster in r(e.g., faster package for rm Anova, or try a regression approach)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by nbrosowsky** [comment]
How about some sort of divide and conquer strategy? \

![image](https://user-images.githubusercontent.com/24325637/41480759-72074346-709d-11e8-9ad5-da2d7fa61d4d.png)

![image](https://user-images.githubusercontent.com/24325637/41480770-7f137eb0-709d-11e8-824c-903f8114a196.png)

![image](https://user-images.githubusercontent.com/24325637/41480786-86dee3d2-709d-11e8-892e-bf7bdfaff18d.png)

![image](https://user-images.githubusercontent.com/24325637/41480794-8d9cd1c0-709d-11e8-8304-98a3422d3f05.png)

**Comment by CrumpLab** [comment]
Maria D’Angelo just tweeted a suggestion that we  “try a linear mixed effects model with lme4 if you go down the regression route.”

So we should try that too

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by CrumpLab** [comment]
I like those divide and conquer graphs, I hadn't thought about going that direction.

**Comment by CrumpLab** [comment]
One solution I've found is to use the rm.anova function from the Rfast package. I had to add some code to the function because it was only reporting F and p, but we also want MSE. This works, and we can use it for an omnibus test.

**Comment by CrumpLab** [comment]
A related approach:

For each subject we have 45 mean IKSIs across the position and word length variables. We want to answer the question: do we have evidence that there are differences among these means that are not due to chance. The omnibus one-way repeated measures ANOVA tells gives us an insane F-value (> 200), and non-existent p-value, so we can say there are differences not likely to have been produced by chance.

The omnibus test does not tell us where the differences are. We could conduct individual comparisons for that. One very conservative approach is compute all possible comparisons. There are 990 possible unique comparisons for 45 means. Doing all of these tests increases likelihood of type I error, so we can use the bonferroni correction (.05/990 ~= .0000505) for our alpha. I conducted all of the possible paired sample t-tests, used the correct alpha, then produced this plot of the absolute difference scores between means (I used absolute so it would print better). The dark gray squares indicate NOT significant, and the light grey squares indicate Significant. This is a comprehensive way of saying that most of the individual condition means are different from one another, and that the differences are not due to chance. 

<img width="458" alt="screen shot 2018-06-19 at 10 40 19 am" src="https://user-images.githubusercontent.com/4038982/41604418-34bdc452-73ad-11e8-984c-5336cb1743a7.png">

**Comment by nbrosowsky** [comment]
Yeah, I'm struggling with this one. 

There's so much data here that almost everything is always significant. I ran some linear mixed models and it works. I compared different models:

    model1 = lmer(mean_IKSI ~ let_pos * word_lengths + (1|Subject), data = subject_means, REML=FALSE)
    model2 = lmer(mean_IKSI ~ let_pos + word_lengths + (1|Subject), data = subject_means, REML=FALSE)
    anova(model1,model2)

The model with the interaction is better than the model without. Everything is significant.

But, what I keep thinking about is the inconsistencies between ISKIs and uncertainty. It looks like a good fit.. except for a couple positions. While the first position is off, we might expect that. What's troubling is how far off the third letter position is (and to a lesser degree, the late letter position slowing). Uncertainty peaks on the third letter position, but IKSI is really low. If you look at different word lengths the trend is the same: uncertainty peaks earlier than IKSIs.

So I'm wondering: Are we using the right H values? We're only taking into account letter position uncertainty ignoring conditional letter probabilities (probability N given N-1, etc.). There are bigram frequencies in each position which we could also use to calculate H for any given position.

**Comment by CrumpLab** [comment]
Glad to see some dabbling with lme4. I still need to bone up on how those models work.

I'm going to close this issue for now, and at the same time open a new issue on explaining variation in mean IKSI with letter uncertainty #17 

Reason for closing: I completed a repeated measures analysis of mean IKSI as a function of position and word length, and added it to the results section of the paper. I did an omnibus test, which showed there was significant differences between the means, and I added all possible comparisons between means showing which differences were significant (bonferroni corrected). For now, this provides strong statistical support for the observation that mean IKSI does vary by position and word length, replicating Ostry (1983). The next problem is to determine whether variation in letter uncertainty by position and word length explains the variation in mean IKSI by position and word length. 

Nick raises some important issues about how to calculate H that we should discuss in #17.
